exchanges/ftx/ftx.go

```diff
@@ -0,0 +1,785 @@
+	getFundingPayments       = "/funding_payments"
+	getLeveragedTokens       = "/lt/tokens"
+	getTokenInfo             = "/lt/%s"
+	getLTBalances            = "/lt/balances"
+	getLTCreations           = "/lt/creations"
+	requestLTCreation        = "/lt/%s/create"
+	getLTRedemptions         = "/lt/redemptions"
+	requestLTRedemption      = "/lt/%s/redeem"
+	getListQuotes            = "/options/requests"
+	getMyQuotesRequests      = "/options/my_requests"
+	createQuoteRequest       = "/options/requests"
+	deleteQuote              = "/options/requests/%s"
+	getQuotesForQuote        = "/options/requests/%s/quotes"
+	createQuote              = "/options/requests/%s/quotes?"
+	getMyQuotes              = "/options/my_quotes"
+	deleteMyQuote            = "/options/quotes/%s"
+	acceptQuote              = "/options/quotes/%s/accept"
+	getOptionsInfo           = "/options/account_info"
+	getOptionsPositions      = "/options/positions"
+	getPublicOptionsTrades   = "/options/trades"
+	getOptionsFills          = "/options/fills"
+	ftxRateInterval          = time.Minute
+	ftxRequestRate           = 180
+
+	// Other Consts
+	stopOrderType         = "stop"
+	trailingStopOrderType = "trailingStop"
+	takeProfitOrderType   = "takeProfit"
+	buy                   = "buy"
+	sell                  = "sell"
+	newStatus             = "new"
+	openStatus            = "open"
+	closedStatus          = "closed"
+	limitOrder            = "limit"
+	marketOrder           = "market"
+	defaultTime           = "0001-01-01 00:00:00 +0000 UTC"
+	spotString            = "spot"
+	futuresString         = "future"
+)
+
+// Start implementing public and private exchange API funcs below
+
+// GetMarkets gets market data
+func (f *FTX) GetMarkets() (Markets, error) {
+	var resp Markets
+	return resp, f.SendHTTPRequest(ftxAPIURL+getMarkets, &resp)
+}
+
+// GetMarket gets market data for a provided asset type
+func (f *FTX) GetMarket(marketName string) (Market, error) {
+	var market Market
+	return market, f.SendHTTPRequest(fmt.Sprintf(ftxAPIURL+getMarket, marketName),
+		&market)
+}
+
+// GetOrderbook gets orderbook for a given market with a given depth (default depth 20)
+func (f *FTX) GetOrderbook(marketName string, depth int64) (OrderbookData, error) {
+	strDepth := strconv.FormatInt(depth, 10)
+	var resp OrderbookData
+	var tempOB TempOrderbook
+	err := f.SendHTTPRequest(fmt.Sprintf(ftxAPIURL+getOrderbook, marketName, strDepth), &tempOB)
+	if err != nil {
+		return resp, err
+	}
+	resp.MarketName = marketName
+	for x := range tempOB.Result.Asks {
+		resp.Asks = append(resp.Asks, OData{Price: tempOB.Result.Asks[x][0],
+			Size: tempOB.Result.Bids[x][1],
+		})
+	}
+	for y := range tempOB.Result.Bids {
+		resp.Bids = append(resp.Bids, OData{Price: tempOB.Result.Bids[y][0],
+			Size: tempOB.Result.Bids[y][1],
+		})
+	}
+	return resp, nil
+}
+
+// GetTrades gets trades based on the conditions specified
+func (f *FTX) GetTrades(marketName, startTime, endTime string, limit int64) (Trades, error) {
```

Comment:
If you change startTime and endTime to be time.Time you can use built in methods IsZero to check its valid and formatint from the strconv package

```go
	if !startTime.IsZero() {
		params.Set("start_time", strconv.FormatInt(startTime.Unix(), 10))
	}

	if !endTime.IsZero() {
		params.Set("end_time", strconv.FormatInt(endTime.Unix(), 10))
	}
	if endTime.Before(startTime) {
		return resp, errors.New("startTime cannot be after than endTime")
	}
```

This can be done for all params that are time based